modules/camera/camera_picamera2.py

```diff
@@ -0,0 +1,71 @@
+"""
+Picamera2 implementation of the camera wrapper.
+"""
+
+import numpy as np
+
+# Picamera2 library only exists on Raspberry Pi
+try:
+    import picamera2
+except ImportError:
+    pass
+
+from . import base_camera
+
+
+class CameraPiCamera2(base_camera.BaseCameraDevice):
+    """
+    Class for the Picamera2 implementation of the camera.
+    """
+
+    __create_key = object()
+
+    @classmethod
+    def create(cls, width: int, height: int) -> "tuple[True, CameraPiCamera2] | tuple[False, None]":
+        """
+        Picamera2 Camera.
+
+        width: Width of the camera.
+        height: Height of the camera.
+
+        Return: Success, camera object.
+        """
+        try:
+            camera = picamera2.Picamera2()
+
+            config = camera.create_still_configuration(
+                {"size": (width, height), "format": "RGB888"}
```

Comment:
Can you add parameters for "ExposureTime" and "AnalogueGain"? We will need to set these for the IR camera detection